internal/pkg/agent/install/perms_unix.go

```diff
@@ -18,19 +19,26 @@ func fixPermissions(topPath string) error {
 	return recursiveRootPermissions(topPath)
 }
 
-func recursiveRootPermissions(path string) error {
-	return filepath.Walk(path, func(name string, info fs.FileInfo, err error) error {
-		if err == nil {
-			// all files should be owned by root:root
-			err = os.Chown(name, 0, 0)
-			if err != nil {
-				return err
-			}
-			// remove any world permissions from the file
-			err = os.Chmod(name, info.Mode().Perm()&0770)
-		} else if errors.Is(err, fs.ErrNotExist) {
+func recursiveRootPermissions(root string) error {
```

Comment:
Correct me if I'm wrong but the changes to this function (and some other minor changes in this PR) are unrelated to the actual bug being fixed?  In the future, could you please pull such refactoring/cleanup changes into their own PR please? It makes things easier to review and, should a revert be needed, it makes it easier to revert just the relevant changes.